### PR TITLE
PLANNER-1674 Change values to cover more branches and add comments to AndCompositeTerminationTest

### PR DESCRIPTION
**Pull request by michaltomco**
Changed test cases that were suppose to cover corner cases but after consulting SonarCloud did not. Add negative cases of time gradient. In addition, add explaining comments, so that the choice of values is more transparent.

### CODE REVIEW

File: optaplanner-core/src/test/java/org/optaplanner/core/impl/solver/termination/AndCompositeTerminationTest.java [hunk 1]
```diff
@@ -74,27 +74,33 @@ public void calculateSolverTimeGradientTest() {
...
         when(termination2.calculateSolverTimeGradient(solverScope)).thenReturn(0.5);
+        //min(0.0,0.5) = 0.0
         assertEquals(0.0, compositeTermination.calculateSolverTimeGradient(solverScope), 0.0);
 
-        when(termination1.calculateSolverTimeGradient(solverScope)).thenReturn(0.1);
-        when(termination2.calculateSolverTimeGradient(solverScope)).thenReturn(0.1);
-        assertEquals(0.1, compositeTermination.calculateSolverTimeGradient(solverScope), 0.0);
+        when(termination1.calculateSolverTimeGradient(solverScope)).thenReturn(-1.0);
+        when(termination2.calculateSolverTimeGradient(solverScope)).thenReturn(-1.0);
+        //negative number are unsupported and ignored, min(unsupported,unsupported) = 1.0 (default)
```
Comment (ge0ffrey) [inline optaplanner-core/src/test/java/org/optaplanner/core/impl/solver/termination/AndCompositeTerminationTest.java]: Code style: Normally our comments have a space after // and start with a capital. So "// Negative ..."
